src/uct/ib/base/ib_device.c

```diff
@@ -457,6 +554,15 @@ void uct_ib_device_cleanup(uct_ib_device_t *dev)
 
     ucs_debug("destroying ib device %s", uct_ib_device_name(dev));
 
+    /* TODO uncomment once uct_ep_flush(CANCEL) non blocking done
+     * ucs_assert(kh_size(&dev->async_events_hash) == 0); */
+    kh_destroy_inplace(uct_ib_async_event, &dev->async_events_hash);
```

Comment:
i'd add ucs_debug() message here if non-empty, and after flush(CACNEL) nb is done, change it to ucs_warn